obspy/io/nordic/tests/test_nordic.py

```diff
@@ -0,0 +1,684 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Functions for testing the obspy.io.nordic functions
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA @UnusedWildImport
+
+import numpy as np
+
+import inspect
+import io
+import os
+import unittest
+
+from obspy import read_events, Catalog, UTCDateTime, read
+from obspy.core.event import Pick, WaveformStreamID, Arrival, Amplitude
+from obspy.core.event import Event, Origin, Magnitude
+from obspy.core.event import EventDescription, CreationInfo
+from obspy.clients.fdsn import Client
+from obspy.io.nordic.core import is_sfile, read_spectral_info, read_nordic
+from obspy.io.nordic.core import readwavename, blanksfile, write_nordic
+from obspy.io.nordic.core import nordpick, stationtoseisan, readheader
+from obspy.io.nordic.core import _int_conv, _readheader, _evmagtonor
+from obspy.io.nordic.core import write_select
+from obspy.io.nordic.core import _float_conv, _nortoevmag, _str_conv
+
+
+class TestNordicMethods(unittest.TestCase):
+    """
+    Test suite for nordic io operations.
+    """
+    def setUp(self):
+        self.path = os.path.dirname(os.path.abspath(inspect.getfile(
+            inspect.currentframe())))
+        self.testing_path = os.path.join(self.path, "data")
+
+    def test_read_write(self):
+        """
+        Function to test the read and write capabilities of sfile_util.
+        """
+        # Set-up a test event
+        test_event = full_test_event()
+        # Add the event to a catalogue which can be used for QuakeML testing
+        test_cat = Catalog()
+        test_cat += test_event
+        # Write the catalog
+        test_cat.write("Test_catalog.xml", format='QUAKEML')
+        # Read and check
+        read_cat = read_events("Test_catalog.xml")
+        os.remove("Test_catalog.xml")
+        self.assertEqual(read_cat[0].resource_id, test_cat[0].resource_id)
+        for i in range(len(read_cat[0].picks)):
+            for key in read_cat[0].picks[i].keys():
+                # Ignore backazimuth errors and horizontal_slowness_errors
+                if key in ['backazimuth_errors', 'horizontal_slowness_errors']:
+                    continue
+                self.assertEqual(read_cat[0].picks[i][key],
+                                 test_cat[0].picks[i][key])
+        self.assertEqual(read_cat[0].origins[0].resource_id,
```

Comment:
Maybe you should set `read_ev = read_cat[0]` and `test_ev = test_cat[0]`, and then you don't need to use all those `[0]`.